### ISSUE TITLE
Add DropdownMenu.Overlay

### ISSUE THREAD

**Issue by darrylyeo** [radix-ui/primitives#1679]
### Overview

Implement an `Overlay` component to `DropdownMenu` similar to the one found in `Dialog`.

### Examples in other libraries

This would function similarly to the `Overlay` found in `Dialog` and `AlertDialog`.

### Who does this impact? Who is this for?

Like a dialog, the overlay would help make the dropdown menu content stand out and visually indicate that the user can click outside to dismiss the menu.

**Comment by benoitgrelard** [comment]
Hi @darrylyeo,
It isn't very common to display menus/popover with overlays behind them.
Do you have examples of that in the wild?

If we were to make this happen, they would only be for modal mode, which `Popover` isn't by default, but `DropdownMenu` is. The part would be optional regardless I guess.

Thoughts on this @andy-hook?

**Comment by andy-hook** [comment]
It’s not particularly common but I have seen it used in the wild, amazon comes to mind: 

![CleanShot 2022-09-22 at 09 09 53](https://user-images.githubusercontent.com/11708259/191693926-be86322b-49f9-41f1-9bdc-c7aa9f7790f3.png)

I also remember this being mentioned as a potential convenience in `NavigationMenu`.

> they would only be for modal mode, which Popover isn't by default

Aside but I sometimes wonder if this default on Popover was the right choice from a consistency perspective.

**Comment by darrylyeo** [comment]
Agreed, I think this could be added to `NavigationMenu` and even `ContextMenu` as well!

Plenty of examples in iOS where context menus dim or blur the background:

<img src="https://706e0528e83c6c9a2424-db085549fff54c6dc092d2fc63e195ef.ssl.cf1.rackcdn.com/2019-9-8-16-11-36.png" width="500">
<img src="https://i.stack.imgur.com/mTbrH.jpg" width="250">
<img src="https://i.stack.imgur.com/XR63N.jpg" width="250">

**Comment by emroot** [comment]
I second that. used in plenty of places, would love to see this land <3

**Comment by yudinvictor** [comment]
is there any progress on this?

**Comment by artemshchirov** [comment]
is there any progress on this?

**Comment by jaithirani** [comment]
Is there a workaround for now?

**Comment by ishaqibrahim-conv-cw** [comment]
@jaithirani You can make a custom `<Overlay/>` component and nest it inside the root component. E.g. for `NavigationMenu` here's how I'm doing it:

```
const [isOpen, setIsOpen] = useState<boolean>(false);

return (
  <NavigationMenu
    defaultValue=""
    onValueChange={(val) => (val ? setIsOpen(true) : setIsOpen(false))}
    className={cn(className)}
  >
    {/* other components */}
    {isOpen && <Overlay />}
  </NavigationMenu>
);

```

Overlay:

```
export const Overlay = () => {
  return (
    <div className="absolute left-0 z-0 top-14 h-screen w-full bg-black bg-opacity-20"></div>
  );
};

```

**Comment by kodgurkini** [comment]
Another solution for adding a backdrop would be to simply add a shadow that fills the whole background behind the Popover container, like so:

`<Popover.Content
        style={{
          boxShadow: "0 0 0 12000px rgba(0, 0, 0, 0.32)",
        }}
      >`

**Comment by dcdm3g** [comment]
> It’s not particularly common but I have seen it used in the wild, amazon comes to mind: 
> 
> ![CleanShot 2022-09-22 at 09 09 53](https://user-images.githubusercontent.com/11708259/191693926-be86322b-49f9-41f1-9bdc-c7aa9f7790f3.png)

I need exactly this implementation. For now, I will make the popover a controlled component and create a custom overlay, but it would be nice to have a `Popover.Overlay` component.

**Comment by adityacodepublic** [comment]
``` tsx
type BackdropOptions = "opaque" | "blur" | "transparent";

interface PopoverContentProps
  extends React.ComponentPropsWithoutRef<typeof PopoverPrimitive.Content> {
  backdrop?: BackdropOptions;
}

const PopoverContent = React.forwardRef<
  React.ElementRef<typeof PopoverPrimitive.Content>,
  PopoverContentProps
>(({ className, align = "center", sideOffset = 4, backdrop, ...props }, ref) => (
  <PopoverPrimitive.Portal>
       <div className="relative">
        <div 
          className={cn("fixed inset-0 animate-fade-in",
            backdrop === "opaque" ? "bg-black/25" : backdrop === "blur" ? "bg-black/25 backdrop-blur-sm":"",)}
        />
        <PopoverPrimitive.Content
          ref={ref}
          align={align}
          sideOffset={sideOffset}
          className={cn(
            "z-50 w-72 rounded-md border bg-popover p-4 text-popover-foreground shadow-md outline-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2",
            className
          )}
          {...props}
        />
      </div>
  </PopoverPrimitive.Portal>
))
```

this works 
add backdrop prop to the PopoverContent 
``` tsx
        <PopoverContent className="w-[250px] p-0 " backdrop="blur">
```
